packages/debug/src/browser/debug-variables-source-tree.ts

```diff
@@ -0,0 +1,89 @@
+/********************************************************************************
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, postConstruct } from '@theia/core/shared/inversify';
+import { ExpandableTreeNode, TreeNode } from '@theia/core/lib/browser';
+import { SourceTree, TreeElement, TreeElementNode, TreeElementNodeParent } from '@theia/core/lib/browser/source-tree';
+import { Disposable } from '@theia/core';
+import { DebugScope, DebugVariable } from './console/debug-console-items';
+import { DebugSessionManager } from './debug-session-manager';
+
+@injectable()
+export class DebugVariablesSourceTree extends SourceTree {
+
+    @inject(DebugSessionManager)
+    protected readonly debugSessionsManager: DebugSessionManager;
+
+    @postConstruct()
+    protected init(): void {
+        this.toDispose.push(Disposable.create(() => this.expandedElements.clear()));
+        this.toDispose.push(this.debugSessionsManager.onDidStartDebugSession(() => {
+            this.firstNodeExpanded = true;
+            this.expandedElements.clear();
+        }));
+    }
+
+    protected readonly expandedElements = new Set<string>();
+    protected firstNodeExpanded = true;
+    protected readonly firstNodeId = '__source__:0';
```

Comment:
What is the significance of this value?

Reply:
I've added a comment to explain this.

Reply:
Let me request instead doing the following, now that I've done a search to figure out where the actual value comes from: in `packages/core/src/browser/source-tree/source-tree.ts` l. 133, extract the string '__source__' as `SOURCE_NODE_ID_PREFIX` and from line 45 extract `':'` `SOURCE_NODE_ID_SEPARATOR`, and then you can create this constant as 
```
`${SOURCE_NODE_ID_PREFIX}${SOURCE_NODE_ID_SEPARATOR}0\`
```

Reply:
I've changed this. I've quickly looked through the code and it looks like these two strings are not used anywhere else.